mlir/lib/Dialect/Math/Transforms/ExpandPatterns.cpp

```diff
@@ -202,6 +202,44 @@ static LogicalResult convertCeilOp(math::CeilOp op, PatternRewriter &rewriter) {
   rewriter.replaceOp(op, ret);
   return success();
 }
+
+// Convert `math.fpowi` to a series of `arith.mulf` operations.
+// If the power is negative, we divide one by the result.
+static LogicalResult convertFPowIOp(math::FPowIOp op,
+                                    PatternRewriter &rewriter) {
+  ImplicitLocOpBuilder b(op->getLoc(), rewriter);
+  Value base = op.getOperand(0);
+  Value power = op.getOperand(1);
+  Type baseType = base.getType();
+
+  Attribute cstAttr;
+  if (!matchPattern(power, m_Constant(&cstAttr)))
+    return failure();
+
+  auto iAttr = dyn_cast<SplatElementsAttr>(cstAttr);
+  if (!iAttr)
+    return failure();
+
+  int64_t powerInt = iAttr.getSplatValue<int64_t>();
```

Comment:
Do we want to support the scalar case? `math.fpowi %0, %1 : f32, i64`

Reply:
https://github.com/openxla/iree/issues/16906 The impression I had from the issue was to support tensors instead of scalars, but I can add this case, too.

Reply:
There's not reason not to support scalars and vectors here

Reply:
Added.